File: Python/BinaryTreeTraversals.py
class Node:
    left = None
    right = None
    data = 0
    
    def __init__(self, value: int) -> None:
        self.data = value

class BinaryTree:
    head = None
    def __init__(self):
        pass
    def walkPreOrder(self, current: Node, path):
        if current == None:
            return path
            
        path.append(current.data)
        self.walkPreOrder(current.left, path)
        self.walkPreOrder(current.right, path)
        return path
        
    def preOrderTraversal(self) -> list(int):
        return self.walkPreOrder(self.head, [])
        
    def walkInOrder(self, current: Node, path):
        if current == None:
            return path
            
        self.walkInOrder(current.left, path)
        path.append(current.data)
        self.walkInOrder(current.right, path)
        return path
        
    def inOrderTraversal(self) -> list(int):
        return self.walkInOrder(self.head, [])
        
    def walkPostOrder(self, current: Node, path):
        if current == None:
            return path
            
        self.walkPostOrder(current.left, path)
        self.walkPostOrder(current.right, path)
        path.append(current.data)
        return path
        
    def postOrderTraversal(self) -> list(int):
        return self.walkPostOrder(self.head, [])
        
    def breadthFirstSearch(self, value):
        queue = [self.head]
        while len(queue) != 0:
            current = queue.pop(0)
            if current == None:
                continue

            if current.data == value:
                return True
            
            queue.append(current.left)
            queue.append(current.right)
        
        return False

File: Java/BinaryTreeTraversals.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;

public class BinaryTreeTraversals {
    Node head;

    public static class Node{
        Node left;
        Node right;
        int data;

        public Node(int value){
            data = value;
        }
    }

    public ArrayList<Integer> walkPreOrder(Node current, ArrayList<Integer> path){
        if(current == null){
            return path;
        }

        path.add(current.data);
        walkPreOrder(current.left, path);
        walkPreOrder(current.right, path);
        return path;
    }

    public ArrayList<Integer> preOrderTraversal(){
        return walkPreOrder(head, new ArrayList<Integer>());
    }

    public static ArrayList<Integer> walkInOrder(Node current, ArrayList<Integer> path){
        if(current == null){
            return path;
        }

        walkInOrder(current.left, path);
        path.add(current.data);
        walkInOrder(current.right, path);
        return path;
    }

    public ArrayList<Integer> inOrderTraversal(){
        return walkInOrder(head, new ArrayList<Integer>());
    }

    public ArrayList<Integer> walkPostOrder(Node current, ArrayList<Integer> path){
        if(current == null){
            return path;
        }

        walkPostOrder(current.left, path);
        walkPostOrder(current.right, path);
        path.add(current.data);
        return path;
    }

    public ArrayList<Integer> postOrderTraversal(){
        return walkPostOrder(head, new ArrayList<Integer>());
    }

    public boolean breadthFirstSearch(int value){
        Queue<Node> queue = new LinkedList<Node>();
        queue.add(head);

        while(!queue.isEmpty()){
            Node current = queue.remove();
            if(current == null){
                continue;
            }

            if(current.data == value){
                return true;
            }

            queue.add(current.left);
            queue.add(current.right);
        }

        return false;

    }

}
